types/react/index.d.ts

```diff
@@ -355,6 +356,31 @@ declare namespace React {
     const Suspense: ExoticComponent<SuspenseProps>;
     const version: string;
 
+    interface Interaction {
+        id: number;
+        name: string;
+        timestamp: number;
+    }
+    /**
+     * {@link https://github.com/bvaughn/rfcs/blob/profiler/text/0000-profiler.md#detailed-design | API}
+     */
+    type OnRenderCallback = (
```

Comment:
This is a public type since it's exported out of the namespace. Can we rename this to be more specifically coupled to the profiler, like `ProfilerRenderCallback` or something? Same goes for `Interaction`.

Reply:
Totally agree on `OnRenderCallback`.

I would rather mark the `Interaction` as internal for now. It's technically part of `scheduler/tracing` and should be imported from there once it's available. I guess renaming it to `SchedulerInteraction` and marking it as private with a notice is the safest course of action.

Reply:
If you want it to be private, declare it outside the `React` namespace.